### PR TITLE
throttle calls to TTS

### PR DESCRIPTION
**Pull request by railsagainstignorance**

### CODE REVIEW

File: bin/lib/fetchContent.js [hunk 1]
```diff
@@ -220,11 +223,17 @@ function articleAsItem(uuid) {
...
+	debug(`articlesAsItems: uuids.length=${uuids.length}, CAPI_CONCURRENCE=${CAPI_CONCURRENCE}`);
+
+	const articlePromisers = uuids.map( uuid => {
+		return function () {
+			return articleAsItem(uuid);
+		};
 	});
+
+	return directly(CAPI_CONCURRENCE, articlePromisers)
+	.then( outputs => outputs.filter( op => {return op !== null;}) )
```
Comment (benbarnett) [inline bin/lib/fetchContent.js]: Up to you but you could use `.then( outputs => outputs.filter(op => op) )` for potential simples

File: bin/lib/queue.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	constructor (items) {
+		this.items = items || [];
+	}
+
+	attachDirectlyInstance (directly) {
+		this.directly = directly;
+	}
+
+	push (func) {
+		this.items.push.apply(this.items, [].slice.call(arguments));
```
Comment (benbarnett) [inline bin/lib/queue.js]: [`Array.from()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) is another way of doing array-things to array-like-things, or could maybe do this? Unless I'm misreading:

```js
push(funcs) {
  this.items.concat(funcs)
}
```

File: bin/lib/directly.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+		} else if (!this.running) {
+			// never take the Promise.all shortcut as even if the initial list is short, it
+			// could easily grow to exceed the concurrence limit.
+			while (this.funcs.length && this.concurrence - this.competitors.length) {
+				this.executeOne();
+			}
+			this.startRace();
+		}
+		this.running === true;
```
Comment (benbarnett) [inline bin/lib/directly.js]: Should this be `this.running = true`?

File: bin/lib/queue.js (hunk 2 above)
Comment (railsagainstignorance) [reply bin/lib/queue.js]: something for Rhys...

File: bin/lib/directly.js (hunk 3 above)
Comment (railsagainstignorance) [reply bin/lib/directly.js]: something for Rhys...

File: bin/lib/fetchContent.js (hunk 1 above)
Comment (railsagainstignorance) [reply bin/lib/fetchContent.js]: good to know, defo shorter, buuut a teensy bit more magic involved.

**Pull request merged**